tasks/secrets.yml

```diff
@@ -3,10 +3,22 @@
   file:
     dest: "{{ _run_temp_dir.path }}/src/{{ item.path | dirname }}"
     state: directory
-  loop: "{{ pantheon_deploy.secrets | default([]) }}"
+  loop: "{{ pantheon_deploy.secrets  | default([]) }}"
+  when: pantheon_deploy.secrets.json is undefined
+- name: Create directories for JSON Secrets
+  file:
+    dest: "{{ _run_temp_dir.path }}/src/{{ item.path | dirname }}"
+    state: directory
+  loop: "{{ pantheon_deploy.secrets.json }}"
 - name: Write secrets
   copy:
     dest: "{{ _run_temp_dir.path }}/src/{{ item.path }}"
     content: "{{ item.value }}"
   loop: "{{ pantheon_deploy.secrets | default([]) }}"
-
+  when: item.value is defined
+- name: Write JSON secrets
+  copy:
+    dest: "{{ _run_temp_dir.path }}/src/{{ item.path }}"
+    content: "{{ item.values | to_nice_json(indent=2, width=80, ensure_ascii=False) }}"
+  loop: "{{ pantheon_deploy.secrets.json | default([]) }}"
+  when: pantheon_deploy.secrets.json is defined
```

Comment:
The conditional here isn't really necessary as the `pantheon_deploy.secrets.json | default([])` loop handles it gracefully.

Reply:
Removed in latest commit, thank you!